### PR TITLE
feat: add state for source manager

### PR DESCRIPTION
**Pull request by shanicky**
## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

as title

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/meta/src/stream/stream_manager.rs [hunk 1]
```diff
@@ -623,9 +612,32 @@ where
     /// Dropping materialized view is done by barrier manager. Check
     /// [`Command::DropMaterializedView`] for details.
     pub async fn drop_materialized_view(&self, table_id: &TableId) -> Result<()> {
+        let table_fragments = self
```
Comment (yezizp2012) [inline src/meta/src/stream/stream_manager.rs:615]: Can we get `source_fragments` directly from a new interface inside FragmentManager? 🤔

File: src/meta/src/stream/stream_manager.rs [hunk 2]
```diff
@@ -623,9 +612,32 @@ where
...
+            .await?;
+
+        let mut source_fragments = HashMap::new();
+        fetch_source_fragments(&mut source_fragments, &table_fragments);
+
         self.barrier_manager
             .run_command(Command::DropMaterializedView(*table_id))
             .await?;
+
+        let mut actor_ids = HashSet::new();
```
Comment (yezizp2012) [inline src/meta/src/stream/stream_manager.rs:627]: Ditto.

**Review by yezizp2012 — COMMENTED** [review]
Rest LGTM.

File: src/meta/src/stream/stream_manager.rs (hunk 1 above)
Comment (shanicky) [reply src/meta/src/stream/stream_manager.rs:615]: It doesn't seem to work, the table fragments are passed in as external parameters when creating materialized view, `fetch_source_fragments` is already the maximum level of abstraction

File: src/meta/src/stream/stream_manager.rs (hunk 2 above)
Comment (shanicky) [reply src/meta/src/stream/stream_manager.rs:627]: Using Default::default here would make it impossible to derive the type

Comment (shanicky) [reply src/meta/src/stream/stream_manager.rs:627]: image

**Review by tabVersion — APPROVED** [review]

**Pull request merged**